src/pipeline/rackhd/vagrant/Build.groovy

```diff
@@ -0,0 +1,61 @@
+package pipeline.rackhd.vagrant
+
+def buildBoxFromOVF(String target_dir, String cache_project_name, String os_ver, String rackhd_version, String rackhd_apt_repo, String rackhd_dir){
+    String cache_image_dir = "$WORKSPACE/cache_image"
+    // Copy ovf from the artifact of project VAGRANT_CACHE_BUILD
+    step ([$class: 'CopyArtifact',
```

Comment:
I'm afraid it's better to put this (copy Artifact) from buildBoxFromOVF to caller (buildBox())
this function can focus on what it should do

Reply:
Thanks!
Fixed.